### PR TITLE
feat:[SSCA-580]: Added sort and search in policy violations listing API

### PR DESCRIPTION
**Pull request by arpit-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52278)

### CODE REVIEW

**Review by deepharness — DISMISSED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52278)* status: :shipit: complete! ✓

**Comment by arpit-harness** [comment]
trigger uts

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -184,6 +184,11 @@ paths:
         - $ref: '#/components/parameters/Order'
         - $ref: '#/components/parameters/Page'
         - $ref: '#/components/parameters/Sort'
+        - schema:
+            type: string
+          in: query
+          name: searchText
```
Comment (mankrit-singh) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:190]: casing-> search_text

**Review by mankrit-singh — COMMENTED** [review]
small comment

**Review by mankrit-singh — APPROVED** [review]

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger messagecheck

**Review by deepharness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52278)* status: 2 file(s) to review. 1 change requests. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/EnforcementResultServiceImpl.java` line 81 at r3](https://reviewable.io/reviews/harness/harness-core/52278#-NdAx4PU-WPRmgJwFQaC:-NdAx4PU-WPRmgJwFQaD:bg47tcc) ([raw file](https://github.com/harness/harness-core/blob/6fe80e572f62586bd3d52a9735fcb060724e7324/ssca-manager/service/src/main/java/io/harness/ssca/services/EnforcementResultServiceImpl.java#L81)):*
> ```Java
>     Criteria searchCriteria = new Criteria();
>     Pattern pattern = Pattern.compile("[.]*" + searchText + "[.]*");
>     if (Objects.nonNull(searchText) && !searchText.isEmpty()) {
> ```

always do a negation for if statement , it will make the code looks cleaner. Put the if at top if search text is null or empty no need to go to repo.

**Review by deepharness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52278)* status: :shipit: complete! ✓

**Comment by arpit-harness** [comment]
trigger messagecheck

**Comment by arpit-harness** [comment]
trigger ut7

**Comment by arpit-harness** [comment]
trigger ut7

**Comment by arpit-harness** [comment]
trigger ut0

**Pull request merged**